capstone-rs/examples/recursive.rs

```diff
@@ -0,0 +1,98 @@
+use std::env;
+use std::process;
+use std::fs;
+use std::collections::{HashMap, VecDeque};
+
+use object::{Object, ObjectSection};
+use capstone;
+use capstone::Insn;
+use capstone::prelude::*;
+
+fn main(){
+	let cs = if let Ok(cs) = Capstone::new()
+		.x86()
+		.mode(arch::x86::ArchMode::Mode64)
+        .detail(true)
+		.build() {
+		cs
+	}else{
+		eprintln!("failed to create capstone handle");
+		process::exit(-1);
+	};
+	
+	let args: Vec<_> = env::args().collect();
+	if args.len() != 2 {
+		eprintln!("Usage: {} <file>", args[0]);
+		process::exit(-1);
+	}
+
+	let buf = if let Ok(bd) = fs::read(&args[1]) {
+		bd
+	} else {
+		eprintln!("cannot read file");
+		process::exit(-1);
+	};
+
+	let obj = if let Ok(od) = object::File::parse(&*buf){
+		od
+	}else{
+		eprintln!("cannot parse file");
+		process::exit(-1);
+	};
+
+	let (sec_addr, sec_text) = if let Some(section) = obj.section_by_name(".text") {
+		if let Ok(data) = section.data() {
+			(section.address(), data)
+		}else{
+			eprintln!("cannot get data from .text section");
+			process::exit(1);
+		}
+	}else{
+		eprintln!("no section .text found");
+		process::exit(2);
+	};
+
+    println!(".text section addr: {:#02x?} size: {:#02x?}", sec_addr, sec_text.len());
+    
+    let mut addr_queue: VecDeque<u64> = VecDeque::new();
+    let mut addr_seen: HashMap<u64, bool> = HashMap::new();
+
+    let cs_ins = cs.iter_malloc();
```

Comment:
I *think* this is undefined behavior (instantiating an Rust struct with initialized memory).

To work with uninitialized data, look into the [MaybeUninit](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html) type.

Reply:
I will double check this. However the iter_malloc does only initialize the capstone handle, which is a raw pointer, not a struct:
`csh: *mut c_void` .